tests/app/questionnaire/test_questionnaire_manager.py

```diff
@@ -44,20 +44,75 @@ def test_add_answer_updates_answer_store(self):
         # Then
         self.questionnaire_manager.update_questionnaire_store.assert_called_with('block')
 
-    def test_remove_answer_detaches_answer_from_question(self):
-        answer_to_remove = self.answers[1]
+    def test_remove_answer_single_answer(self):
+        # Given
+        should_remove = mock_answer('answer', 0, self.question)
+        answers = [should_remove]
+        self.questionnaire_manager.state.get_answers = MagicMock(return_value=answers)
+        self.question.remove_answer = Mock()
+
+        # When
+        self.questionnaire_manager.remove_answer('block', self.answer_store, 0)
+
+        # Then
+        self.question.remove_answer.assert_called_with(should_remove)
+
+    def test_remove_answer_single_answer_multiple_instances(self):
+        # Given
+        answers = [
+            mock_answer('answer', 0, self.question),
+            mock_answer('answer', 1, self.question),
+            mock_answer('answer', 2, self.question),
+        ]
+        self.questionnaire_manager.state.get_answers = MagicMock(return_value=answers)
+        self.question.remove_answer = Mock()
+
+        # When
+        self.questionnaire_manager.remove_answer('block', self.answer_store, 1)
+
+        # Then
+        self.question.remove_answer.assert_called_with(answers[1])
 
-        with patch.object(self.question, 'remove_answer') as mock:
-            self.questionnaire_manager.remove_answer('block', self.answer_store, 1)
-            mock.assert_called_with(answer_to_remove)
+    def test_remove_answer_multiple_answer_single_instance(self):
+        # Given
+        answers = [
+            mock_answer('first', 0, self.question),
+            mock_answer('middle', 0, self.question),
+            mock_answer('last', 0, self.question)
+        ]
+        self.questionnaire_manager.state.get_answers = MagicMock(return_value=answers)
+        self.question.remove_answer = Mock()
+
+        # When
+        self.questionnaire_manager.remove_answer('block', self.answer_store, 0)
 
-    def test_remove_answer_removes_from_answer_store(self):
-        answer_to_remove = self.answers[1]
-        flattened_answer = answer_to_remove.flatten()
+        # Then
+        calls = [
+            call(answers[0]), call(answers[1]), call(answers[2])
+        ]
+        self.question.remove_answer.assert_has_calls(calls)
 
-        with patch.object(self.answer_store, 'remove') as mock:
-            self.questionnaire_manager.remove_answer('block', self.answer_store, 1)
-            mock.assert_called_with(flattened_answer)
+    def test_remove_answer_multiple_answer_multiple_instances(self):
```

Comment:
This test name is a bit misleading. I was expecting it to remove multiple answers across multiple instances